accepted/0000-prometheus-tls.md

```diff
@@ -0,0 +1,268 @@
+## Prometheus
+
+In the default configuration it is presumed that untrusted users have access to
+the Prometheus HTTP endpoint. They have access to all time series information
+contained in the database, plus a variety of operational/debugging
+information [[1]](#1).
+
+Applications needing access to Prometheus API:
+
+* _Grafana_
+
+Users needing access to Prometheus API/UI:
+
+* monitoring administrator
+
+## Exporters
+
+Exporters generally only talk to one configured client instance with a preset
+set of commands/requests, which cannot be expanded via their HTTP endpoint.
+Still, all exposed metrics are available to untrusted users.
+
+> Uyuni should optionally support and automate setting up secure
+> configuration of Prometheus and exporters.
+
+# Detailed design
+
+[design]: #detailed-design
+
+## Prometheus
+
+Prometheus 2.24.0 has introduced TLS encryption as well as basic and
+client-certificate authentication [[2]](#2). All available configuration options are
+described in official documentation [[3]](#3). Basic authentication credentials are
+stored in the configuration file. Passwords are hashed with _bcrypt_.
+
+To follow current SUSE Manager security model [[4]](#4) we should consider
+implementing following default configuration:
+
+* Upgrade Prometheus to version >= 2.24.1.
+* [Optional] Create intermediate certificate authority (CA) dedicated for monitoring.
+* [Optional] Create and deploy Prometheus server certificate signed by this CA.
+* Configure `tls_server_config` using this certificate.
+* Configure basic authentication for administrator account.
+
+All these steps should be implemented in Salt prometheus-formula. Alternatively
+certificate generation and deployment could be moved to a new dedicated
+certificates formula.
+
+To simplify deployment, Prometheus administrator account credentials should be set
+statically in the prometheus-formula.
+
+## Node exporter
+
+Node exporter supports the same configuration options as Prometheus server in
+terms of encryption and authentication [[5]](#5). All available options are
+described in exporter-toolkit documentation [[6]](#6).
+
+To follow current SUSE Manager security model we should consider encrypting all
+the communication between node exporters and Prometheus server. Regarding
+authentication, client certificate authentication seems to be the preferred
+approach as the certificate will be anyway available on the Prometheus server.
+It would involve following configuration steps:
+
+* [Optional] Create and deploy CA signed server certificates for all monitored minions.
+* Configure `tls_server_config` in node exporter using these certificates.
+* Configure `tls_config.ca_file` in Prometheus job configuration to use server
+  certificate [[7]](#7).
+* Configure `tls_config` in Prometheus job configuration to use Prometheus
+  server certificate and key files for client cert authentication.
+
+Alternatively to last point basic authentication can be configured. Please note
```

Comment:
Do we plan to support cert authentication, basic authentication or both when it comes to the node exporter?

Reply:
The plan is to go with cert authentication for _node exporter_. Basic authentication is mentioned as an alternative which we could implement if it turns out that it's requested by users or if handling of certificates turns out to be too complex.